### PR TITLE
Fix DROP database

### PR DESCRIPTION
**Pull request by m1ghtym0**
Fixes `mysql_rm_db_internal` to not rely on directories and files on disk.

### CODE REVIEW

File: storage/rocksdb/ha_rocksdb.h [hunk 1]
```diff
@@ -1075,6 +1075,11 @@ bool rocksdb_frm_write(char *frm_path, const uchar *frm, size_t len);
 bool rocksdb_frm_read(const char *frm_path, uchar **frm, size_t *len);
 bool rocksdb_frm_delete(const char *frm_path);
 std::vector<std::string> rocksdb_frm_discover(const char *frm_path);
+int rocksdb_discover_table_names(handlerton *hton
+                                        __attribute__((unused)),
+                                        LEX_CSTRING *db,
+                                        MY_DIR *dir __attribute__((unused)),
```
Comment (thomasten) [inline storage/rocksdb/ha_rocksdb.h]: is `__attribute__((unused))` needed? And if so, I think one usually uses it only at functions definitions, not at declarations.

File: sql/sql_db.cc [hunk 2]
```diff
@@ -890,16 +874,38 @@ mysql_rm_db_internal(THD *thd, const LEX_CSTRING *db, bool if_exists, bool silen
...
-			  ER_DB_DROP_EXISTS, ER_THD(thd, ER_DB_DROP_EXISTS),
+        ER_DB_DROP_EXISTS, ER_THD(thd, ER_DB_DROP_EXISTS),
                           db->str);
       error= false;
       goto update_binlog;
     }
   }
 
-  if (find_db_tables_and_rm_known_files(thd, dirp, dbnorm, db_path, &tables))
+  del_dbopt(opt_path); // Remove dboption hash entry
```
Comment (thomasten) [inline sql/sql_db.cc:884]: What's the effect of moving this after the exists check?

Comment (m1ghtym0) [reply sql/sql_db.cc:884]: We can't delete the db before calling `exists`.
The `del_dbopt` call is not important here and is just logically grouped together with `myrocks::rocksdb_db_delete(opt_path)`

File: storage/rocksdb/ha_rocksdb.h (hunk 1 above)
Comment (m1ghtym0) [reply storage/rocksdb/ha_rocksdb.h]: Not sure if it is needed, I've copied that from other storage implementations not operating on the filesystem.

**Review by thomasten — APPROVED** [review]

**Pull request merged**